andela/ah-backend-prime
# #164798297 Users can follow each other

**What does this PR do?**
Creates functionality for users following each other

**Description of tasks to be completed.**

- Creation of functionality of users to follow each other

- Creation of functionality of users to unfollow each other

**How should this be manually tested?**

- After cloning the repo, cd into it and run the following commands to activate the virtual environment
   `virtualenv venv`
   `source venv/bin/activate`

- Checkout the branch with the functionality of users folllowing/unfollowing eachother
   `git checkout ft-users-follow-each-other-164798297`

- Install all requirements
   `pip install -r requirements.txt`

- Source the environment variables
   `source .env`

- Run the migrations
   `python manage.py makemigrations`
   `python manage.py migrate`

- Run the server
   `python manage.py runserver`

- To test out this functionality one has to first register on the application, in case of a successful registration, they will receive an account activation link on their email, this activation link would then be pasted in the URL section of postman and sent in order to activate and log into ones account where they would get a token which they would use in the Authorization section to access all other routes including the follows route.

- To follow a user, one has to send a **POST** request to the route:
  `http://127.0.0.1:8000/api/v1/profiles/{username}/follow` (see Figure 1)

<img width="1219" alt="Screen Shot 2019-04-10 at 10 22 42" src="https://user-images.githubusercontent.com/26630001/55859400-ab531f80-5b7a-11e9-8508-0ceef22b83c4.png">

**Figure 1** shows successful following of a user anyati after which the response _You are now following anyati_ is returned.

- To unfollow a user, one has to send a **DELETE** request to the route:
   `http://127.0.0.1:8000/api/v1/profiles/{username}/follow` (see Figure 2)

<img width="1219" alt="Screen Shot 2019-04-10 at 10 27 11" src="https://user-images.githubusercontent.com/26630001/55859695-3e8c5500-5b7b-11e9-84f4-5cc172666fb5.png">

**Figure 2** shows successful unfollowing of a user anyati after which the response _You have unfollowed anyati_ is returned.

**What are the relevant pivotal tracker stories?**
   [#164798297](https://www.pivotaltracker.com/story/show/164798297)
